sdk/client/src/response.rs

```diff
@@ -114,8 +119,8 @@ impl MethodResponse {
             MethodResponse::GetNetworkStatus(_) => Method::GetNetworkStatus,
             MethodResponse::GetStateProof(_) => Method::GetStateProof,
             MethodResponse::GetAccountStateWithProof(_) => Method::GetAccountStateWithProof,
-            MethodResponse::GetTransactionsWithProofs => Method::GetTransactionsWithProofs,
-            MethodResponse::GetEventsWithProofs => Method::GetEventsWithProofs,
+            MethodResponse::GetTransactionsWithProofs(_) => Method::GetTransactionsWithProofs,
+            MethodResponse::GetEventsWithProofs(_) => Method::GetEventsWithProofs,
```

Comment:
Rather than explicitly mapping via match, can we not implement trait that matches the method to the response, and call that? ie: MethodResponse::Submit implements MethodPair that'll return the Method::Submit


Outside the scope of this PR.

Reply:
Ah, we need to mold the arguments, with `Submit` being the lone exception.

Reply:
`MethodResponse::Submit(_)` is a variant of the `MethodResponse` enum, you cannot implement a trait on a variant of an enum, only the enum itself so that wouldn't make very much sense to even use a trait for in this instance.